### PR TITLE
chore(image): Add non-root litmus-go image with reduced size and refactor Makefile

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

- This PR reduces litmus go-runner image size to 150Mb.

- Refactor of Makefile to use ENVs for building and pushing images.

### CODE REVIEW

**Comment by ksatchit** [comment]
We need changes here: 
- pulling docker cli
- Adding non-root steps

**Comment by uditgaurav** [comment]
Tested  on both ARM64 and AMD464 both of these:
- [X] Docker cli
- [X] Alpine as base image.
- [ ] non root tests

@ksatchit

File: Makefile [hunk 1]
```diff
@@ -5,10 +5,16 @@
 # Internal variables or constants.
 # NOTE - These will be executed when any make target is invoked.
 #
-IS_DOCKER_INSTALLED = $(shell which docker >> /dev/null 2>&1; echo $$?)
 
+# Docker info
+DOCKER_REPO ?= uditgaurav
```
Comment (ksatchit) [inline Makefile:10]:
```suggestion
DOCKER_REPO ?= litmuschaos
```

File: Makefile [hunk 2]
```diff
@@ -5,10 +5,16 @@
...
+DOCKER_IMAGE ?= go-runner
+DOCKER_TAG ?= non-root
```
Comment (ksatchit) [inline Makefile:12]:
```suggestion
DOCKER_TAG ?= nonroot
```

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**